src/views/v3/Bridge/AssetPicker/index.tsx

```diff
@@ -0,0 +1,557 @@
+          {props.isSource ? (
+            <AmountInput
+              value={amountInput}
+              debauncedValue={debouncedAmountInput}
+              receiveAmount={receiveAmount}
+              supportedSourceTokens={props.tokenList || []}
+              tokenBalance={
+                props.token ? props.balances[props.token.key]?.balance : null
+              }
+              warning={props.amountValidation?.warning}
+              error={props.amountValidation?.error}
+              onChange={handleAmountChange}
+              onDebouncedChange={handleDebouncedAmountChange}
+            />
+          ) : (
+            <Box
+              sx={{
+                display: 'flex',
+                alignContent: 'center',
+                alignItems: 'center',
+                width: '100%',
+                maxWidth: '250px',
+                height: '50px',
+              }}
+            >
+              <TextField
+                fullWidth
+                disabled
+                placeholder="0"
+                slotProps={{
+                  htmlInput: {
+                    style: {
+                      fontSize: '36px',
+                      height: '36px',
+                    },
+                  },
+                  input: {
+                    disableUnderline: true,
+                  },
+                }}
+                variant="standard"
+                value={receiveAmount}
+              />
+            </Box>
+          )}
+          <Card
+            sx={[
+              styles.selector,
+              props.isTransactionInProgress && styles.disabled,
+            ]}
+            data-testid={props.dataTestId}
+            variant="elevation"
+            onMouseDown={(e) => {
+              if (mobile) {
+                setIsDrawerOpen(true);
+              } else {
+                popupState.open(e);
+              }
+            }}
+            onTouchEnd={(e) => {
+              e.stopPropagation();
+              e.preventDefault();
+              if (mobile) {
+                setIsDrawerOpen(true);
+              } else {
+                popupState.open(e);
+              }
+            }}
+            {...triggerProps}
+          >
+            <CardContent sx={styles.cardContent}>
+              <Typography sx={styles.chainSelector} component={'div'} gap={1}>
+                <AssetBadge chainConfig={chainConfig} token={props.token} />
+                {selection}
+              </Typography>
+            </CardContent>
+          </Card>
+        </Box>
+        {props.isSource && (
+          <Box
+            sx={{
+              height: '24px',
+              display: 'flex',
+              flexDirection: 'row',
+              justifyContent: 'space-between',
+              alignItems: 'center',
+              gap: '8px',
+            }}
+          >
+            <Box>{balance}</Box>
+            <Box>{percentButtons}</Box>
+          </Box>
+        )}
+      </Box>
+      {mobile ? (
```

Comment:
A good example here for breakdown would be that you have 2 components.

Using arbitrary names, `<MobileAssetPicker>` and `<DesktopAssetPicker>` and this code in this file would just be `mobile ? <MobileAssetPicker> : <DesktopAssetPicker>`